core/src/main/scala/org/http4s/ResponseCookie.scala

```diff
@@ -130,17 +130,30 @@ class RequestCookieJar(private val headers: Seq[Cookie])
 }
 
 // see http://tools.ietf.org/html/rfc6265
-final case class Cookie(
-    name: String,
-    content: String,
+sealed abstract class Cookie(val name: String, val content: String) extends Renderable
```

Comment:
I think the file should still be called `Cookie.scala`.

Reply:
I'm personally more partial to:

```scala
sealed trait Cookie extends Renderable{
   def name: String
   def content: String
   ...
}
```
Mostly being convention I see a lot in the scala world, for what in this sense we're defining to be an interface specifically.

Reply:
Where in the API would we continue to abstract over `Cookie`?  Do we even need a base at all?

Reply:
I echo the questioning of the need to actually abstract over `Cookie`. Also, per https://www.reddit.com/r/scala/comments/3887az/everything_you_ever_wanted_to_know_about_sealed/crtvpm5/, it seems wrong to me, albeit allowed in Scala, to use an ADT's sub-type in a signature.

Reply:
Please note that I respectfully question the need. I realize that I've made no commits to this code-base. I look forward to learning from maintainers and authors, yielding to them on this PR!

Reply:
And I appreciate the help!

What I would do is see what doesn't compile if `Cookie` goes away.  Definitely in the `RequestCookieJar`, which would be for `RequestCookie`.  There might be a few more references in the tests.  If everything is obviously related to a `Request` or a `Response`, fix it.  If something serves in both roles, maybe then we have a reason for the base `Cookie` to remain.  But I doubt that this case exists.

Reply:
To be fair @kevinmeredith, I agree with you, but I say merely for the sake of convention, approaching Cookie like an interaface rather than an `ADT` which we will exhaustively match on, given this is scala, not haskell, wherein the subtypes themselves are types.

If this were Haskell, there would be no inheritance whatsoever, simply two record types, as we don't want to work ever in terms of `Cookie`.

IMHO we should get rid of inheriting from a cookie trait altogether, given that whether it's an abstract class or a trait, we can still exhaustively pattern match on `Cookie` like if it was an ADT if we are ever presented with it, so it's of no real difference to use an abstract class over a trait in this case. 

I'm just being a pedant maybe,  but if we're going for this, I'm thinking we should stop people from creating signatures in terms of `Cookie` in the first place.